nifi-toolkit/nifi-toolkit-assembly/src/main/resources/bin/cli.sh

```diff
@@ -111,8 +111,7 @@ run() {
    export NIFI_TOOLKIT_HOME="$NIFI_TOOLKIT_HOME"
 
    umask 0077
-   "${JAVA}" -cp "${CLASSPATH}" ${JAVA_OPTS:--Xms128m -Xmx256m} org.apache.nifi.toolkit.cli.CLIMain "$@"
-   return $?
+   exec "${JAVA}" -cp "${CLASSPATH}" ${JAVA_OPTS:--Xms128m -Xmx256m} org.apache.nifi.toolkit.cli.CLIMain "$@"
```

Comment:
Does this have the same behavior in terms of the exit code that will be returned from calling cli.sh?

I'm not familiar with exec, but it is import for the CLI that exit code from the Java process is returned from calling cli.sh. 

The man page for exec says:
"If there are no redirection errors, the return status is zero; otherwise the return status is non-zero."

It is not clear to me if a redirection error is an error from the Java process, or an error replacing the current shell with the Java process.

Reply:
exec is typically used for redirecting file descriptors for the current shell, the redirection errors should refer to that. Otherwise the return code of the program is not modified, since exec gets out of the way as soon as the program is started so that the return code is not manipulated in any way.

```sh
# start a container 
docker run --rm -ti debian bash

# setup a file to exit with the input argument
cat <<EOF > test.sh
#!/bin/bash

exit \$1
EOF
chmod +x test.sh

# observe that exit codes are preserved
for i in 0 1 2 3; do sh -c "exec ./test.sh $i"; echo $?; done
```